lib/ansible/plugins/callback/nrdp.py

```diff
@@ -0,0 +1,175 @@
+            description: service where the passive check is linked to
+            required: True
+            env:
+                - name : NRDP_ANSIBLE_SERVICENAME
+'''
+
+import os
+
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
+
+from ansible.plugins.callback import CallbackBase
+
+
+class CallbackModule(CallbackBase):
+    '''
+    send ansible-playbook to Nagios server using nrdp protocol
+    '''
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'notification'
+    CALLBACK_NAME = 'nrdp'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    # Nagios states
+    OK = 0
+    WARNING = 1
+    CRITICAL = 2
+    UNKNOWN = 3
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+
+        if not HAS_REQUESTS:
+            self._display.warning(
+                "The required python requests library is not installed."
+                " The NRDP callback plugin is disabled.")
+            self.disabled = True
+
+        self.url = os.getenv("NRDP_URL")
+        if not self.url.endswith('/'):
+            self.url += '/'
+        self.token = os.getenv("NRDP_TOKEN")
+        self.hostname = os.getenv("NRDP_ANSIBLE_HOSTNAME")
+        self.servicename = os.getenv("NRDP_ANSIBLE_SERVICENAME")
+
+        if (self.url or self.token or self.hostname or
+                self.servicename) is None:
+            self._display.warning("NRDP callback wants the NRDP_URL,"
+                                  " NRDP_TOKEN, NRDP_ANSIBLE_HOSTNAME,"
+                                  " NRDP_ANSIBLE_SERVICENAME"
+                                  " environment variables'."
+                                  " The NRDP callback plugin is disabled.")
+            self.disabled = True
+
+        self.play = None
+
+    def _send_nrdp(self, state, msg):
+        '''
+        nrpd service check send XMLDATA like this:
+        <?xml version='1.0'?>
+            <checkresults>
+                <checkresult type='service'>
+                    <hostname>somehost</hostname>
+                    <servicename>someservice</servicename>
+                    <state>1</state>
+                    <output>WARNING: Danger Will Robinson!|perfdata</output>
+                </checkresult>
+            </checkresults>
+        '''
+        xmldata = "<?xml version='1.0'?>\n"
+        xmldata += "<checkresults>\n"
+        xmldata += "<checkresult type='service'>\n"
+        xmldata += "<hostname>%s</hostname>\n" % self.hostname
+        xmldata += "<servicename>%s</servicename>\n" % self.servicename
+        xmldata += "<state>%d</state>\n" % state
+        xmldata += "<output>%s</output>\n" % msg
+        xmldata += "</checkresult>\n"
+        xmldata += "</checkresults>\n"
+
+        try:
+            # disable urllib3 and ssl warnings
+            if hasattr(requests.packages.urllib3, 'disable_warnings'):
+                requests.packages.urllib3.disable_warnings()
+            response = requests.post(
```

Comment:
Consider `open_url` from `ansible.module_utils.urls` as an option to remove the dependency on `requests`.